apis/app_handler.go

```diff
@@ -37,15 +38,15 @@ type CFAppRepository interface {
 
 type AppHandler struct {
 	logger      logr.Logger
-	serverURL   string
+	serverURL   url.URL
 	appRepo     CFAppRepository
 	buildClient ClientBuilder
 	k8sConfig   *rest.Config // TODO: this would be global for all requests, not what we want
 }
 
 func NewAppHandler(
 	logger logr.Logger,
-	serverURL string,
+	serverURL url.URL,
```

Comment:
I'm confused about this change. Why take a URL if we're only going to immediately turn it back into a string again? Why not do call `.URL' at the call site if we happen to have the value in that form? This also seems to complicate the tests, since we have to create a `url.URL` out of the string, just so the constructor can immediately turn it back into a string.